src/lib/swapper/swappers/ArbitrumBridgeSwapper/getTradeQuote/getTradeQuote.ts

```diff
@@ -14,6 +15,23 @@ import { v4 as uuid } from 'uuid'
 import { fetchArbitrumBridgeSwap } from '../utils/fetchArbitrumBridgeSwap'
 import { assertValidTrade } from '../utils/helpers'
 
+export type GetEvmTradeQuoteInputWithWallet = Omit<GetEvmTradeQuoteInput, 'supportsEIP1559'> & {
+  wallet: HDWallet
+}
+
+export const getTradeQuoteWithWallet = async (inputWithWallet: GetEvmTradeQuoteInputWithWallet) => {
+  const { wallet, ...input } = inputWithWallet
+  const supportsEIP1559 = supportsETH(wallet) && (await wallet.ethSupportsEIP1559())
+
+  return getTradeQuote({
+    ...input,
+    sellAsset: input.sellAsset,
+    buyAsset: input.buyAsset,
+    accountNumber: input.accountNumber,
+    supportsEIP1559,
+  })
+}
+
```

Comment:
The second guts of this - this is our queryFn, just not named like so, and living in a sane place (cc @NeOMakinG).
Note that as opposed to the queryKey, it takes the input it's supposed to, without having to do any optional business - by the time we call this, we have narrowed things down and will call this with the right args.
If we don't, TS will yell at us, as it should. Yay type safety!